Assets/HoloToolkit/Input/Scripts/Interactions/TapToPlace.cs

```diff
@@ -123,86 +152,99 @@ private void HandlePlacement()
         {
             if (IsBeingPlaced)
             {
-                SetLayerRecursively(transform, useDefaultLayer: false);
-                InputManager.Instance.AddGlobalListener(gameObject);
+                StartPlacing();
+            } else
+            {
+                StopPlacing();
+            }
+        }
 
-                // If the user is in placing mode, display the spatial mapping mesh.
-                if (AllowMeshVisualizationControl)
-                {
-                    SpatialMappingManager.Instance.DrawVisualMeshes = true;
-                }
+        private void StartPlacing()
+        {
+            var layerCacheTarget = PlaceParentOnTap ? ParentGameObjectToPlace : gameObject;
+            layerCacheTarget.SetLayerRecursively(IgnoreRaycastLayer, out layerCache);
+            InputManager.Instance.PushModalInputHandler(gameObject);
```

Comment:
This is awesome.  In fact, we probably don't even have to change the layers of the object or its children because of this change.